matrix_sdk/src/client.rs

```diff
@@ -1748,6 +1842,67 @@ mod test {
         {}
     }
 
+    #[tokio::test]
+    #[allow(irrefutable_let_patterns)]
+    async fn room_search_all() {
+        let homeserver = Url::from_str(&mockito::server_url()).unwrap();
+
+        let _m = mock(
+            "GET",
+            Matcher::Regex(r"^/_matrix/client/r0/publicRooms".to_string()),
+        )
+        .with_status(200)
+        .with_body_from_file("../test_data/public_rooms.json")
+        .create();
+
+        let client = Client::new(homeserver).unwrap();
+
+        if let get_public_rooms::Response { chunk, .. } = client
+            // .get_public_rooms(Some(10), None, Some(&mockito::server_url().to_string()))
```

Comment:
Is this needed for something?

Reply:
Funny-ish story I was testing and testing trying to get this test to work... This one is a GET request and the other is a POST I had them both as posts :facepalm:. I just forgot to remove one of my many attempts to fix it before I realized it was the wrong request type.